### ISSUE TITLE
Compatibiliy for TB +60

### ISSUE THREAD

**Issue by Frank-Steiner** [IagoSRL/TBX-Unified-Search#36]
Hi,
any chance this addon is released for TB 60 and later?

cu,
Frank

**Comment by KIMWW** [comment]
Me too I'd like to see it working on Thunderbird 60.

**Comment by mlazdans** [comment]
Wondering if this project is abandoned?

**Comment by KIMWW** [comment]
Unfortunately! Yes, the developer has abandoned the project. It has not been updated since 2017...

**Comment by Frank-Steiner** [comment]
Depending on what you needed US for it can be reached with the quick filter bar. I used US because it was just a small input field and not the horrible GUI of the quick filter bar was just taking way too much space for me.
With userChrome.css and a small webextension it's possible to style the quick filter bar so that it looks pretty much like the US input field, moved into the main button bar, and (like in the old days of TB 2) changing "Sender" into "Recipient" when selecting the "Sent" folder.
It must be adjusted a little by anyone who wants to use it because it's tailored very much to my needs, but if someone is not interested in all the additional features that US had but only in styling the quick filter bar into a small, US-like search field, I can post the css stuff and webextension. It will look like this: [pic1](https://www.bio.ifi.lmu.de/~steiner/us1.jpg)  [pic2](https://www.bio.ifi.lmu.de/~steiner/us2.jpg)

**Comment by mlazdans** [comment]
Yeah, I just started to use this separate filter, not that big of a deal indeed.

**Comment by IagoSRL** [comment]
Hello world!

Yes, is not compatible, there is a long time that I had no time to keep progress on this extension.
I just tried it with latest TB 60.7 but there are a lot of things that have broken, even more than when I tried close to a year ago (sorry for not posting updates about progress -or lack of).

I found myself not using Thunderbird for several years now, because of several problems I was having, plus I'm using only my GMail account now and the webapp makes a faster experience for me -especially when searching.

Despite that, would be happy to have time to keep Unified Search updated (and give TB another try) but is not being easy with continuous breaking changes, lack of documentation on the internal features used ([here the official list, nothing about filter/search widgets](https://wiki.mozilla.org/Thunderbird/Add-ons_Guide_57)), and there is an important concern about if any work now will be sent to trash with the advent of WebExtensions and being mandatory or legacy/overlays getting restricted ([as planned on next major version](https://developer.thunderbird.net/add-ons/tb68)).

---------------

I should add, as commented somewhere already, that the approach implementing this extension is a bit "hacky", relying heavily on overlays and existent widgets rather than creating a new one from scratch ([the idea for a cleaner approach is around since 2013](https://github.com/IagoSRL/TBX-Unified-Search/issues/22)). I'm aware that relying a lot on modifying existing widgets is highly problematic since any update at core makes a potential breaking change (while is the 'classic' way for add-ons).

This is because the conception of this extension was to be an internal proof of concept of what I posted [at a Bugzilla comment](https://bugzilla.mozilla.org/show_bug.cgi?id=526221#c13), that I end publishing and adding features to test ideas of combining search and filtering ([some](https://bugzilla.mozilla.org/show_bug.cgi?id=570815#c42) more [related](https://bugzilla.mozilla.org/show_bug.cgi?id=516543) "bugs" [here](https://bugzilla.mozilla.org/show_bug.cgi?id=520040)). It ended becoming somewhat popular, but still didn't get traction from developers that rejected the idea on any of the approaches I showed viable (from the multiple settings enabled at the extension, which doesn't encourages to have all that combinations at TB, but just a testing area of which ones make a better user experience, with the rest getting deleted once is clear).

If someone has the time and/or knowledge to make it compatible with latest versions, please let me know or send PRs!! Or if you end creating/finding a viable alternative as a WebExtension, I would like to heard about it and deprecate this in favor of that.
Onno Ekker (@oeekker) already helped to make previous version compatible with TB 57, which I appreciate a lot.

Cheers,
Iago Lorenzo Salgueiro
